### PR TITLE
Chaincode

### PR DESCRIPTION
**Pull request by n-bhasin**
I created the chaincode. plz review the code and let me know

### CODE REVIEW

**Comment by n-bhasin** [comment]
I modified the code. but I have a doubt while updating the medical info. And I created a separate function for removing consent.

File: chaincode/javascript/lib/medical.js [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+        this.medicalId = this.medicalId + 1;
+        return true;
+    }
+
+    //add doctors under consentTo
+
+    /**
+     *
+     * @param {Context} ctx
+     * @param {string} patientRecordId
```
Comment (majid-T) [inline chaincode/javascript/lib/medical.js:81]: Just checking that this is patientId correct?

File: chaincode/javascript/lib/medical.js [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+            consentTo,
+        } = medicalRecordObj;
+        const recordAsBytes = await ctx.stub.getState(patientId);
+        if (!recordAsBytes || recordAsBytes.length === 0) {
+            throw new Error(`${patientId} does not exist`);
+        }
+        //converting into string
+        const record = JSON.parse(recordAsBytes.toString());
+
+        //checking for permission
```
Comment (majid-T) [inline chaincode/javascript/lib/medical.js:163]: I checked the assignment document and there was no restriction on adding data. I guess when you are Health Entity you can add for any user.

File: chaincode/javascript/lib/medical.js [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+        record.medicalRecord[medicalRecordId].dateUploaded = now.toFormat(
+            "YYYY/MM/DD PP HH:MI"
+        );
+        record.medicalRecord[
+            medicalRecordId
+        ].medicalRecordData = medicalRecordData;
+
+        await ctx.stub.putState(patientId, Buffer.from(JSON.stringify(record)));
+
+        return true;
```
Comment (majid-T) [inline chaincode/javascript/lib/medical.js:186]: Return medicalRecordId to be used in front end as confirmation.

File: chaincode/javascript/package.json [hunk 4]
```diff
@@ -0,0 +1,48 @@
+{
+    "name": "fabcar",
```
Comment (majid-T) [inline chaincode/javascript/package.json:2]: try changing the name. But if not possible its not a big deal.

File: chaincode/javascript/package.json [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    "version": "1.0.0",
+    "description": "FabCar contract implemented in JavaScript",
```
Comment (majid-T) [inline chaincode/javascript/package.json:4]: Try changing contract name here too if possible.

**Comment by n-bhasin** [comment]
I'll create a new one

**Pull request closed without merging**